app/models/user.rb

```diff
@@ -1,13 +1,56 @@
 class User < Base
+  include AASM
+
   has_many :organisation_users
   has_many :organisations, through: :organisation_users
 
-  custom_endpoint :accept_transition_screen, on: :member, request_method: :patch
-  custom_endpoint :accept_rollover_screen, on: :member, request_method: :patch
   custom_endpoint :accept_terms, on: :member, request_method: :patch
 
-  def self.member(id)
-    new(id: id)
+  def self.member(id, state)
+    new(id: id, state: state)
+  end
+
+  aasm do
+    state :new, initial: true
+    state :transitioned
+    state :rolled_over
+    state :notifications_configured
+
+    event :accept_transition_screen do
+      transitions from: :new, to: :transitioned
+    end
+
+    event :accept_rollover_screen do
+      transitions from: :transitioned, to: :rolled_over do
+        guard { Settings.rollover }
+      end
+    end
+
+    event :accept_notifications_screen do
+      transitions from: :rolled_over, to: :notifications_configured do
+        guard { associated_with_accredited_body }
+        guard { !notifications_configured }
+      end
+    end
+  end
+
+  def next_state
+    aasm
+        .states(permitted: true)
+        .map(&:name)
+        .first
+  end
+
+  def aasm_read_state(_name = :default)
+    state.try(:to_sym).presence || self.class.aasm.initial_state
```

Comment:
Feels like without the context of the discussion we had this morning, it could be a bit obscure why we have to do this bit. Can we add a comment here just to explain why we have to accommodate `nil` here?